coderd/workspaces_test.go

```diff
@@ -1563,6 +1563,10 @@ func TestWorkspaceWatcher(t *testing.T) {
 	}()
 
 	wait("agent connected")
+	// This could be racy since we don't guarantee that agent all
+	// lifecycles are reported (if they happen in quick succession).
+	wait("agent lifecycle starting")
+	wait("agent lifecycle ready")
```

Comment:
Updated retry surfaced this issue. We should look into refactoring this test to be more reliable, e.g. by verifying why we got the updated workspace state.

Reply:
Ok, improved this slightly by checking agent statuses.